ros/ros1/OdometryServer.hpp

```diff
@@ -42,15 +44,19 @@ class OdometryServer {
     /// Register new frame
     void RegisterFrame(const sensor_msgs::PointCloud2::ConstPtr &msg);
 
+    /// If user ask, report the pose in the given child_frame
+    Sophus::SE3d CloudToBaseTf(const std::string &pointcloud_frame_id) const;
+
     /// Ros node stuff
     ros::NodeHandle nh_;
     ros::NodeHandle pnh_;
     int queue_size_{1};
 
     /// Tools for broadcasting TFs.
-    tf2_ros::TransformBroadcaster tf_broadcaster_;
+    std::unique_ptr<tf2_ros::TransformBroadcaster> tf_broadcaster_;
```

Comment:
I'd avoid using pointers here. As far as I know, there is no benefit on ROS 1 (compared to composable nodes in ROS 2). So, this "complicates" the implementation for no good reason (as I see it now).  Again, please challenge any bs**it I say :) Happy to change my mind al;ways